lib/name.mli

```diff
@@ -27,11 +27,20 @@ open Cstruct
 type label
 
 (** Domain name, as a list of strings. *)
-type domain_name = string list
+type domain_name
```

Comment:
Would be nice to just call this `t` as we're breaking the interface anyway. `Name.domain_name` is a bit verbose.